src/main/java/address/updater/DependencyHistoryHandler.java

```diff
@@ -0,0 +1,118 @@
+            return;
+        }
+
+        Optional<List<String>> libraries = ManifestFileReader.getLibrariesInClasspathFromManifest();
+
+        if (!libraries.isPresent()) {
+            logger.debug("No libraries being used - should not happen");
+            assert false;
+            return;
+        }
+
+        updateVersionDependencies(currentVersion, libraries.get());
+    }
+
+    /**
+     * Updates the dependencies of a version of the application
+     */
+    public void updateVersionDependencies(Version version, List<String> verDependencies) {
+        dependenciesForVersionsInUse.put(version, verDependencies);
+        writeVersionDependencyHistory();
+    }
+
+    /**
+     * Gets dependencies for every version known
+     */
+    public HashMap<Version, List<String>> getDependenciesOfAllVersion() {
+        return dependenciesForVersionsInUse;
+    }
+
+    /**
+     * Deletes dependencies of version which are no longer used
+     */
+    public void cleanUpUnusedDependencyVersions(List<Version> unusedVersions) {
+        Iterator<Map.Entry<Version, List<String>>> it = dependenciesForVersionsInUse.entrySet().iterator();
+
+        while (it.hasNext()) {
+            Map.Entry<Version, List<String>> entry = it.next();
+            if (unusedVersions.contains(entry.getKey())) {
+                logger.debug("Removing {}", entry.getKey());
+                it.remove();
+            }
+        }
+
+        writeVersionDependencyHistory();
+    }
+
+    private void writeVersionDependencyHistory() {
+        if (!DEPENDENCY_HISTORY_FILE.exists()) {
+            try {
+                FileUtil.createFile(DEPENDENCY_HISTORY_FILE);
+            } catch (IOException e) {
+                logger.debug("Failed to create dependency file", e);
+            }
+        }
+
+        try {
+            StorageManager.serializeObjectToJsonFile(DEPENDENCY_HISTORY_FILE, dependenciesForVersionsInUse);
+        } catch (JsonProcessingException e) {
+            logger.debug("Failed to convert dependencies to JSON", e);
+        } catch (IOException e) {
+            logger.debug("Failed to write dependencies to file", e);
+        }
+    }
+
+    private void loadVersionDependencyHistory() {
+        if (!DEPENDENCY_HISTORY_FILE.exists()) {
+            logger.debug("Dependencies file does not exist yet");
+            return;
+        }
+
+        try {
+            String json = FileUtil.readFromFile(DEPENDENCY_HISTORY_FILE);
+            dependenciesForVersionsInUse = JsonUtil.fromJsonStringToGivenType(json,
+                    new TypeReference<HashMap<Version, List<String>>>() {});
+        } catch (IOException e) {
+            logger.debug("Failed to read dependencies from file");
+            e.printStackTrace();
+        }
+    }
+
+    public List<String> getCurrentVersionDependencies() {
+        return dependenciesForVersionsInUse.get(currentVersion);
+    }
+}
```

Comment:
I think putting in `Dependencies` object inside `Version` will unnecessarily complicate the `Version` object, since sometimes we use `Version` without any need about `Dependencies`. I will keep it this way for now.